subsys/lorawan/lorawan.c

```diff
@@ -429,14 +455,14 @@ int lorawan_send(uint8_t port, uint8_t *data, uint8_t len, uint8_t flags)
 			mcpsReq.Req.Confirmed.fBuffer = data;
 			mcpsReq.Req.Confirmed.fBufferSize = len;
 			mcpsReq.Req.Confirmed.NbTrials = lorawan_conf_msg_tries;
-			mcpsReq.Req.Confirmed.Datarate = lorawan_datarate;
+			mcpsReq.Req.Confirmed.Datarate = default_datarate;
```

Comment:
Why can't this be `current_datarate`?

Reply:
It doesn't matter whether it is current or default datarate.
If ADR is enabled, this value is ignored.
If ADR is disabled, `current_datarate` is set to `default_datarate` due to the logic at the end of `lorawan_join`.

Reply:
Right, but using `default_datarate` doesn't sound right to me. But anyway, I was wondering if we can just use the single variable for caching the datarate.

Reply:
I can change the name over. As for merging `default_datarate` and `current_datarate`, that depends on what the intended behavior of the following code is:
```
int main(void) {
    lorawan_enable_adr(true);
    lorawan_set_datarate(DR_2);
    lorawan_join(&join_cfg);
    for(int i = 0; i < 50; i++) {
        lorawan_send(data, data_len);
    }
    /* Rejoin the network */
    lorawan_join(&join_cfg);
}
```
If the second join request should use the same datarate that we set originally, we need to preserve that value.
Keep in mind that the second join could be to a completely different network, with different link budgets etc.
I don't actually want to join multiple networks, but other people might.

Reply:
> If the second join request should use the same datarate that we set originally, we need to preserve that value.
Keep in mind that the second join could be to a completely different network, with different link budgets etc.
I don't actually want to join multiple networks, but other people might.

Hmm, makes sense. Then please use `current_datarate` for `lorawan_send` and add a note above the declaration of `default_datarate` on why we have 2 variables. Otherwise LGTM!